### PR TITLE
fix: Embedded postgres file owner fix

### PR DESCRIPTION
**Pull request by sum35h**
Fixes https://github.com/appsmithorg/appsmith/issues/24358 where postgres system files owner is updated with mongodb user on updating Appsmith.

Tested manually by updating v1.9.22 to nightly.

### CODE REVIEW

File: deploy/docker/entrypoint.sh [hunk 1]
```diff
@@ -349,15 +347,16 @@ init_postgres() {
...
-      mkdir -p $POSTGRES_DB_PATH
+      mkdir -p "$POSTGRES_DB_PATH"
 
       # Postgres does not allow it's server to be run with super user access, we use user postgres and the file system owner also needs to be the same user postgres
-      chown postgres:postgres $POSTGRES_DB_PATH
+      chown postgres:postgres "$POSTGRES_DB_PATH"
 
       # Initialize the postgres db file system
-      su -m postgres -c "/usr/lib/postgresql/13/bin/initdb -D $POSTGRES_DB_PATH"
+      su postgres -c "/usr/lib/postgresql/13/bin/initdb -D $POSTGRES_DB_PATH"
```
Comment (sharat87) [inline deploy/docker/entrypoint.sh:359]: Can you detail why we're making this change please?

File: deploy/docker/entrypoint.sh [hunk 2]
```diff
@@ -31,8 +31,6 @@ if [[ -n "${FILESTORE_IP_ADDRESS-}" ]]; then
   FILE_SHARE_NAME="$(echo "$FILE_SHARE_NAME" | xargs)"
 
   echo "Running appsmith for cloudRun"
-  echo "creating mount point"
-  mkdir -p "$stacks_path"
```
Comment (sharat87) [inline deploy/docker/entrypoint.sh:35]: Can you detail why this isn't needed anymore for CloudRun please?

**Review by sharat87 — COMMENTED** [review]
There's a couple of changes that aren't just setting the owner of Postgres files, can you confirm they're related to the same problem, if yes, share details, and if not, can we revert and keep the hotfix small and focused please?

File: deploy/docker/entrypoint.sh (hunk 2 above)
Comment (sum35h) [reply deploy/docker/entrypoint.sh]: Missed that, uas on stale release.

**Pull request merged**